sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/EventGridPublisherAsyncClient.java

```diff
@@ -70,17 +74,26 @@ public EventGridServiceVersion getServiceVersion() {
      * @return the completion.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Void> sendEvents(Iterable<EventGridEvent> events) {
+    public Mono<Void> sendEventGridEvents(Iterable<EventGridEvent> events) {
         if (events == null) {
             return monoError(logger, new NullPointerException("'events' cannot be null."));
         }
-        return withContext(context -> sendEvents(events, context));
+        return withContext(context -> sendEventGridEvents(events, context));
     }
 
-    Mono<Void> sendEvents(Iterable<EventGridEvent> events, Context context) {
+    Mono<Void> sendEventGridEvents(Iterable<EventGridEvent> events, Context context) {
         final Context finalContext = context != null ? context : Context.NONE;
         return Flux.fromIterable(events)
```

Comment:
Add null check, this will throw NPE if  `events` is null.

Reply:
Moved null check to internal methods from public methods, which calls the internal methods.